File: Java/1518.java
class Solution {
    public int numWaterBottles(int numBottles, int numExchange) {
        int result = 0;
        while (numBottles >= numExchange) {
            result += numExchange * (numBottles / numExchange);
            numBottles = numBottles - numExchange * (numBottles / numExchange) + (numBottles / numExchange);
        }
        return result + numBottles;
    }
}

File: Python/1518.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        empty_bottles, full_bottles = 0, numBottles
        drank = 0

        while full_bottles > 0:
            drank += full_bottles
            empty_bottles += full_bottles
            full_bottles = empty_bottles // numExchange
            empty_bottles = empty_bottles % numExchange

        return drank
